tsl/src/fdw/data_node_scan_plan.c

```diff
@@ -302,13 +306,32 @@ force_group_by_push_down(PlannerInfo *root, RelOptInfo *hyper_rel)
 	groupexprs = get_sortgrouplist_exprs(root->parse->groupClause, root->parse->targetList);
 	new_partnatts = list_length(groupexprs);
 
-	/* Only reallocate the partitioning attributes array if it is smaller than
+	/* Only reallocate the partitioning attributes arrays if it is smaller than
 	 * the new size */
 	if (partscheme->partnatts < new_partnatts)
+	{
+		partopfamily = palloc0(new_partnatts * sizeof(Oid));
+		partopcintype = palloc0(new_partnatts * sizeof(Oid));
+		partcollation = palloc0(new_partnatts * sizeof(Oid));
+		nullable_partexprs = palloc0(new_partnatts * sizeof(List *));
+		for (i = 0; i < partscheme->partnatts; ++i)
+		{
+			partopfamily[i] = partscheme->partopfamily[i];
+			partopcintype[i] = partscheme->partopcintype[i];
+			partcollation[i] = partscheme->partcollation[i];
+			nullable_partexprs[i] = hyper_rel->nullable_partexprs[i];
+		}
+		partscheme->partopfamily = partopfamily;
+		partscheme->partopcintype = partopcintype;
+		partscheme->partcollation = partcollation;
+		hyper_rel->nullable_partexprs = nullable_partexprs;
+
```

Comment:
This is what `realloc` does, so why do it explicitly? Also note that you can declare variables inside the if-body, which avoids the need to reset the `i` below and avoid a potential bug.

Reply:
Realloc doesn't zero out the free space, which is a problem cause we probably need the `InvalidOid` and `NULL` pointers in there.

Reply:
In that case, I think that `palloc0` and `memcpy` is a better alternative and easier to maintain, but I leave the choice to you. Either way, you need a comment about why you do it this way, otherwise, somebody will "optimize" this and introduce a subtle bug.